packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit.tsx

```diff
@@ -1,5 +1,302 @@
+          value: InboundShipmentRow[K]
+        ) => {
+          const batch = inboundItem.batches[id];
+          if (inboundItem && batch) {
+            batch[key] = value;
+            setInboundItem({
+              ...inboundItem,
+              batches: { ...inboundItem.batches, [id]: batch },
+            });
+          }
+        },
+      };
+
+      setInboundItem({ ...inboundItem });
+    }
+  };
+
+  useEffect(() => {
+    if (item) setInboundItem({ ...item });
+    else setInboundItem(item);
+  }, [item]);
+
+  return (
+    <>
+      <ModalRow>
+        <ModalLabel label={t('label.item')} />
+        <Grid item flex={1}>
+          <ItemSearchInput
+            currentItemName={item?.itemName}
+            onChange={onChangeItem}
+          />
+        </Grid>
+      </ModalRow>
+      {item && (
+        <ModalRow>
+          <Grid style={{ display: 'flex', marginTop: 10 }} flex={1}>
+            <ModalLabel label={t('label.code')} />
+            <BasicTextInput
+              disabled
+              sx={{ width: 150 }}
+              value={item.itemCode}
+            />
+          </Grid>
+          <Grid
+            style={{ display: 'flex', marginTop: 10 }}
+            justifyContent="flex-end"
+            flex={1}
+          >
+            <ModalLabel label={t('label.unit')} justifyContent="flex-end" />
+            <BasicTextInput
+              disabled
+              sx={{ width: 150 }}
+              value={item.itemUnit}
+            />
+          </Grid>
+        </ModalRow>
+      )}
+      <Divider margin={5} />
+      {inboundItem && (
+        <TableContainer sx={{ height: 400 }}>
+          <Table>
+            <TableHead>
+              <TableRow>
+                <HeaderCell>{t('label.batch')}</HeaderCell>
+                <HeaderCell>{t('label.num-packs')}</HeaderCell>
+                <HeaderCell>{t('label.pack-size')}</HeaderCell>
+
+                <HeaderCell>{t('label.sell')}</HeaderCell>
+                <HeaderCell>{t('label.cost')}</HeaderCell>
+                <HeaderCell>Line Total</HeaderCell>
+                <HeaderCell>Units</HeaderCell>
+                <HeaderCell>{t('label.expiry')}</HeaderCell>
+                <HeaderCell>% Margin</HeaderCell>
+
+                <HeaderCell>
+                  Volume/
+                  <br />
+                  Pack
+                </HeaderCell>
+                <HeaderCell>
+                  Weight/
+                  <br />
+                  Pack
+                </HeaderCell>
+                <HeaderCell>Location</HeaderCell>
+                <HeaderCell>
+                  Sent # <br /> Packs
+                </HeaderCell>
+                <HeaderCell>
+                  Sent Pack <br /> Size
+                </HeaderCell>
+              </TableRow>
+            </TableHead>
+
+            {flattenInboundItems([inboundItem]).map(batch => (
+              <BatchRow key={batch.id} batch={batch} />
+            ))}
+
+            <Fab
```

Comment:
The rectangular drop shadow on the FAB troubles me:

<img width="50" alt="Screen Shot 2021-11-22 at 12 01 49 PM" src="https://user-images.githubusercontent.com/9192912/142782383-87b511fe-23c3-4db7-aa6d-2d9c6d0ef9d2.png">

Changing the viewbox of `PlusCircleIcon` (almost) fixes the alignment, and then you can use the `40px` version of the fab, which I think looks good:

`viewBox="0 0 21 20.25"`

<img width="83" alt="Screen Shot 2021-11-22 at 12 18 33 PM" src="https://user-images.githubusercontent.com/9192912/142782862-a3be9455-b052-4b06-99a1-508e30909093.png">

that's with this on the fab: 
```
              sx={{
                alignSelf: 'flex-end',
                margin: '10px',
                alignItems: 'center',
                justifyContent: 'center',
              }}
```

Reply:
much better! Thanks